src/poetry/repositories/legacy_repository.py

```diff
@@ -212,9 +206,19 @@ def client_cert(self) -> Path | None:
 
     @property
     def authenticated_url(self) -> str:
-        if not self._session.auth:
-            return self.url
+        """
+        Set authentication credentials and return a URL with basic auth
```

Comment:
This code path is only used for uploads not for requests etc. see Authenticator for details.